### PR TITLE
perf: add liveness and readiness probes to deployments

### PR DESCRIPTION
**Pull request by dimakis**
[linked here in the codeflare operator](https://github.com/project-codeflare/codeflare-operator/pull/150)

### CODE REVIEW

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -38,14 +38,13 @@ spec:
         livenessProbe:
           httpGet:
             path: /healthz
-            port: 8081
```
Comment (astefanutti) [inline config/manager/manager.yaml:41]: Is changing the default port number required?

File: deployment/deployment.yaml [hunk 2]
```diff
@@ -25,4 +25,15 @@ spec:
           requests:
             cpu: "2"
             memory: "2G"
+          livenessProbe:
```
Comment (astefanutti) [inline deployment/deployment.yaml:28]: I think we need to assess what's the purpose of that `deployment` directory first. It seems redundant with the Kustomize manifests.

Comment (astefanutti) [reply deployment/deployment.yaml:28]: Yes, it should be removed, but I see it's used in the README. We should update the later to rely on Kustomize manifests, and rely on `make deploy`.

Comment (dimakis) [reply deployment/deployment.yaml]: this suggestion has been added 👍

**Review by astefanutti — APPROVED** [review]

**Comment by dimakis** [comment]
@anishasthana @KPostOffice can one of ye give me a lgtm here so I can merge please?

**Comment by asm582** [comment]
kind bump to review this PR :)

**Comment by asm582** [comment]
do we know what is pending for getting this PR merged?

**Comment by dimakis** [comment]
@KPostOffice sorry to poke again, but can you approve this?

**Comment by asm582** [comment]
Can we please make sure we have manually tested this PR?

**Pull request merged**